src/Widgets/DriverRow.vala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * Authored by: Leonhard Kargl <leo.kargl@proton.me>
+ */
+
+public class About.DriverRow : Gtk.ListBoxRow {
+    private static Gtk.SizeGroup button_size_group = new Gtk.SizeGroup (HORIZONTAL);
+
+    public signal void install ();
+
+    public string driver_name { get; construct; }
+    public bool installed { get; construct; }
+
+    public DriverRow (string driver_name, bool installed) {
+        Object (driver_name: driver_name, installed: installed);
+    }
+
+    construct {
+        var icon = new Gtk.Image.from_icon_name ("application-x-firmware") {
+            pixel_size = 32
+        };
+
+        var label = new Gtk.Label (driver_name) {
+            hexpand = true,
+            xalign = 0
+        };
+
+        var install_button = new Gtk.Button.with_label (installed ? _("Installed") : _("Install")) {
```

Comment:
I think it might be more appropriate to use a radio here to make it more clear that this is a mutually exclusive selection